### PR TITLE
Use kurl version from spec if provided

### PR DESCRIPTION
**Pull request by laverya**
#### What type of PR is this?
type::bug

#### What this PR does / why we need it:
This fixes the bug preventing the kurlVersion that is specified in the installer spec from being fully applied. 

#### Which issue(s) this PR fixes:

Fixes #
[SC-42119](https://app.shortcut.com/replicated/story/42119/kurlversion-isn-t-fully-applied-when-it-s-set-as-part-of-the-kurl-manifest)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/controllers/Scripts.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ export class Installers {
 
     response.set("X-Kurl-Hash", installer.hash());
     response.status(200);
-    return this.templates.renderInstallScript(installer, kurlVersion);
+    return this.templates.renderInstallScript(installer, installer.spec.kurl?.installerVersion);
```
Comment (emosbaugh) [inline web/src/controllers/Scripts.ts:121]: wont this make kurlversion in the url no longer work?

Comment (laverya) [reply web/src/controllers/Scripts.ts:121]: no, because of the code change to have the version in the URL inserted into the installer spec

Comment (emosbaugh) [reply web/src/controllers/Scripts.ts:121]: so what takes precedence? what should take precedence?

Comment (garcialuis) [reply web/src/controllers/Scripts.ts:121]: This handler is not used to get installers for custom specs, this just gets the latest installer or a specific version that is specified in the URL. With this change the kurlVersion from the URL will be added into the spec (line 117), at the end of the day these two would be the same: 
`return this.templates.renderInstallScript(installer, kurlVersion);`
`return this.templates.renderInstallScript(installer, installer.spec.kurl?.installerVersion);`

File: web/src/controllers/Scripts.ts [hunk 2]
```diff
@@ -145,6 +145,11 @@ export class Installers {
       response.status(404);
       return notFoundResponse;
     }
+
+    if (!kurlVersion && installer.spec.kurl) {
```
Comment (garcialuis) [inline web/src/controllers/Scripts.ts:149]: @emosbaugh Here's where we use the custom spec that may have the installerVersion. I added this condition to check for the installerVersion in the spec if there wasn't one provided in the URL.

**Comment by laverya** [comment]
Could you add a daily test that 1. specifies an old kurl installer version, and 2. somehow checks that that old version is used?
(maybe grep the install.sh script for something that was removed?)

File: web/src/util/services/templates.ts [hunk 3]
```diff
@@ -136,6 +136,7 @@ export async function manifestFromInstaller(i: Installer, kurlUrl: string, repli
   if (kurlVersion) {
     kurlUtilImage = `replicated/kurl-util:${kurlVersion}`;
     kurlBinUtils = `kurl-bin-utils-${kurlVersion}.tar.gz`;
+    i.spec.kurl = {additionalNoProxyAddresses: [], installerVersion: kurlVersion}
```
Comment (laverya) [inline web/src/util/services/templates.ts:139]:
```suggestion
    if (i.spec.kurl) {
      i.spec.kurl.installerVersion = kurlVersion;
    } else {
      i.spec.kurl = {additionalNoProxyAddresses: [], installerVersion: kurlVersion}
    }
```
that way you don't clobber other kurl settings here

**Review by emosbaugh — COMMENTED** [review]
I feel like it would be valuable to add tests to the handlers in Scripts.ts. im happy to help with that if you need.

File: web/src/util/services/templates.ts (hunk 3 above)
Comment (garcialuis) [reply web/src/util/services/templates.ts]: Good catch, I've added your suggestion. Thank you

**Review by emosbaugh — APPROVED** [review]
This is awesome. So glad to have some controller tests!

**Review by laverya — COMMENTED** [review]
lgtm

**Pull request merged**